java/vector/src/main/java/org/apache/arrow/vector/VarCharVector.java

```diff
@@ -270,6 +273,21 @@ public void setSafe(int index, Text text) {
     setSafe(index, text.getBytes(), 0, text.getLength());
   }
 
+  @Override
+  public int hashCode (int index) {
+    //TODO cal hashCode in memory level
+    byte[] values = get(index);
```

Comment:
There is a memory copy here, which can be removed. Another piece of work is in progress, please see https://github.com/apache/arrow/pull/4844